Support --version option for hnc plugin

tcnksm opened the pull request:
Related: https://github.com/kubernetes-sigs/multi-tenancy/issues/861

Add `cobra.Command.Version`. This enables the plugin to have `--version` option flag to show its version. I'm not sure the way to embed the version string is what you expect.  

## Test

I've build it on local and tried its output,

```bash
$ go build -ldflags="-X sigs.k8s.io/multi-tenancy/incubator/hnc/internal/version.Version=v0.5.2" -o kubectl-hns  cmd/kubectl/main.go
```
```bash
$ ./kubectl-hns version
kubectl-hns version v0.5.2
```

tcnksm commented:
> Instead of --version, how about a version subcommand? That seems to fit in better with kubectl version and go version.

Ok. Let me add the `version` subcommand. I think, as a command-line user, I also expect to have `--version` command as well. So let me keep it. If you think it's not good UI then I will remove it.

tcnksm commented on incubator/hnc/internal/version/version.go:
```diff
@@ -0,0 +1,19 @@
+package version
+
+// Version is verion for HNC. This variable is overwritten when build time.
+var Version = "(UNKNOWN)"
```
I also see the practice to put it on specialized package like this. I think the reason is that version information sometimes needs to be referred to from multiple places. If you prefer to put it on `main` then I'm gonna move it to main and pass it via a global variable to the `kubectl` package.

adrianludwin replied:
For the record: using a separate package sgtm.

adrianludwin commented:
No, leaving it in its own package makes sense. Thanks. I'll have a closer
look tomorrow.

On Mon, Aug 24, 2020 at 9:23 PM Taichi Nakashima <notifications@github.com>
wrote:

> @tcnksm <https://github.com/tcnksm> requested your review on: #1040
> <https://github.com/kubernetes-sigs/multi-tenancy/pull/1040> Support
> --version option for hnc plugin.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes-sigs/multi-tenancy/pull/1040#event-3687796682>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE43PZFM7EETEOY5VBNMGCTSCMHAVANCNFSM4QG5B6IQ>
> .
>

tcnksm commented on incubator/hnc/internal/kubectl/root.go:
```diff
@@ -99,6 +101,7 @@ func init() {
 	rootCmd.AddCommand(newTreeCmd())
 	rootCmd.AddCommand(newCreateCmd())
 	rootCmd.AddCommand(newConfigCmd())
+	// rootCmd.AddCommand(newVersionCmd())
```
Sorry, fixed

adrianludwin commented on incubator/hnc/internal/version/version.go:
```diff
@@ -0,0 +1,19 @@
+
+package version
+
+// Version is verion for HNC. This variable is overwritten when build time.
```
Maybe a bit more information is useful here: "This variable is overwritten in cloudbuild.yaml to match the version of the Docker image."

tcnksm replied:
Yeah! I think it's better. Let me update the comment.

tcnksm commented:
> I think we're almost ready to go. Can you please make two last changes?

Thank you for the instruction. I've squashed commits and updated description 👉  https://github.com/kubernetes-sigs/multi-tenancy/pull/1040/commits/0b15a9bb70477c6f3eb0cf60fad67613bb627b10
